internal/messaging/processor.go

```diff
@@ -183,9 +230,46 @@ func (p *processor) Respond(msg *Message) error {
 	}
 
 	md := &msg.Metadata
-	// rewrite sender & recipient metadata
-	md.Recipient = md.Sender
-	md.Sender = p.userID
+
+	mybotAddress := md.Recipient
+
+	/// TODO: Uncomment after Chewques implementation
+	// No message should be without cheque?
+	if md.Cheques == nil {
+		return fmt.Errorf("%w: %s", ErrMissingCheques, msg.Type)
+	}
+
+	for i := 0; i < len(md.Cheques); i++ {
+		// Get info from cheque
+		// Check if the TO CM-Account is correct
+		if !p.identificationHandler.isMyCMAccount(md.Cheques[0].ToCMAccount) {
+			return fmt.Errorf("incorrect CMAccount")
+		}
+
+		// Lookup the FROM CM-Account which is in the cheque
+		// Verify that the sending bot is part of this FROM
+		isInCmAccount, err := p.identificationHandler.isBotInCMAccount(md.Sender[1:43], md.Cheques[i].FromCMAccount)
+		if err != nil {
+			return fmt.Errorf("%w: %w, %s", ErrCheckingCmAccount, err, md.Cheques[i].FromCMAccount)
+		}
+
+		if !isInCmAccount {
+			return fmt.Errorf("bot %s not part of CMAccount %s", md.Sender[1:43], md.Cheques[i].FromCMAccount)
+		}
+		cmAccount, ok := p.identificationHandler.findCmAccount(md.Sender)
+		if ok {
+			md.Recipient = cmAccount.Hex()
+		} else {
```

Comment:
not sure about that part, maybe that should be done automatically on cheque verification, like, update map with cm-acc->bot pair from cheque
lets leave it as it is for now and may be change in my cheques PR